tests/lib/devicetree/api/src/main.c

```diff
@@ -702,15 +702,25 @@ static void test_phandles(void)
 		     "has gpio 1 pin");
 	zassert_true(DT_PHA_HAS_CELL_AT_IDX(TEST_PH, gpios, 1, flags),
 		     "has gpio 1 flags");
-	zassert_true(DT_PHA_HAS_CELL_AT_IDX(TEST_PH, pha_gpios, 1, pin),
-		      "no pha-gpio 1 pin");
-	/* index 1 only has pin, no flags */
+	/* index 1 has nothing, not even a phandle */
+	zassert_false(DT_PROP_HAS_IDX(TEST_PH, pha_gpios, 1),
+		      "pha-gpio 1 should not exist");
+	zassert_false(DT_PHA_HAS_CELL_AT_IDX(TEST_PH, pha_gpios, 1, pin),
+		      "has pha-gpio 0 pin");
 	zassert_false(DT_PHA_HAS_CELL_AT_IDX(TEST_PH, pha_gpios, 1, flags),
-		      "no pha-gpio 1 flags");
+		      "has pha-gpio 0 flags");
+	/* index 2 only has pin, no flags */
 	zassert_true(DT_PHA_HAS_CELL_AT_IDX(TEST_PH, pha_gpios, 2, pin),
+		      "no pha-gpio 1 pin");
+	zassert_false(DT_PHA_HAS_CELL_AT_IDX(TEST_PH, pha_gpios, 2, flags),
+		      "no pha-gpio 1 flags");
```

Comment:
Several messages in the assertions here are quite confusing and misleading. For example, the above two should refer to index 2 and one of them should say "has" instead of "no", but actually I'm not sure which one, as I'm not sure if those messages are intended to say what was expected or what turned out to be incorrect.

Reply:
> Several messages in the assertions here are quite confusing and misleading.

You know, you're totally right. I think most of the assertions in this file are easy enough to read on their own, so I've added a cleanup commit at the beginning of the series that removes the ones that add nothing, or worse are actively confusing.

Thanks.